src/serializers/comparisons.js

```diff
@@ -47,10 +49,35 @@ addHydrateFrom("$gte", makeNew(is.GreaterThanEqual));
 addHydrateFromValues("$in", makeNew(is.In));
 addHydrateFrom("$lt", makeNew(is.LessThan));
 addHydrateFrom("$lte", makeNew(is.LessThanEqual));
-addHydrateFromValues("$nin", makeNew(is.GreaterThan));
 
+addHydrateFromValues("$all", makeNew(is.All));
 
+var notRemap = {
```

Comment:
might want to add an inline comment about what `notRemap` means.  It took me a while to figure out (had to read other code).

Reply:
I think my difficulty was because `not` sounds like you "don't want to remap" instead of "these are the mapping for a not with XXX as the nested type".

`var subtypeOfNotMap` ? Not much better.

Reply:
It's the opposite of each time. So maybe `oppositeTypeMap`?